docs/document/content/others/faq/_index.cn.md

```diff
@@ -5,73 +5,176 @@ weight = 1
+更多关于alias使用方法请参考[Proxool官网](http://proxool.sourceforge.net/configure.html)。
+
+PS：sourceforge网站需要翻墙访问。
+
+## 11. [分片] ShardingSphere提供的默认分布式自增主键策略为什么是不连续的，且尾数大多为偶数？
 
 回答：
 
-必须确保数据库表中该字段和分片算法该字段类型一致，如：数据库中该字段类型为int(11)，泛型所对应的分片类型应为Integer，如果需要配置为Long类型，请确保数据库中该字段类型为bigint。
+ShardingSphere采用snowflake算法作为默认的分布式自增主键策略，用于保证分布式的情况下可以无中心化的生成不重复的自增序列。因此自增主键可以保证递增，但无法保证连续。
+
+而snowflake算法的最后4位是在同一毫秒内的访问递增值。因此，如果毫秒内并发度不高，最后4位为零的几率则很大。因此并发度不高的应用生成偶数主键的几率会更高。
+
+在3.1.0版本中，尾数大多为偶数的问题已彻底解决，参见：https://github.com/apache/shardingsphere/issues/1617
+
+## 12. [分片] 如何在inline分表策略时，允许执行范围查询操作（BETWEEN AND、\>、\<、\>=、\<=）？
+
+回答：
+
+1. 需要使用4.1.0或更高版本。
+2. 调整以下配置项（需要注意的是，此时所有的范围查询将会使用广播的方式查询每一个分表）：
+- 4.x版本：`allow.range.query.with.inline.sharding`设置为true即可（默认为false）。
+- 5.x版本：在InlineShardingStrategy中将`allow-range-query-with-inline-sharding`设置为true即可（默认为false）。
+
+## 13. [数据加密] JPA 和 数据加密无法一起使用，如何解决？
+
+回答：
+
+由于数据加密的DDL尚未开发完成，因此对于自动生成DDL语句的JPA与数据加密一起使用时，会导致JPA的实体类(Entity)无法同时满足DDL和DML的情况。
+
+解决方案如下：
+
+1. 以需要加密的逻辑列名编写JPA的实体类(Entity)。
+2. 关闭JPA的auto-ddl，如 auto-ddl=none。
+3. 手动建表，建表时应使用数据加密配置的`cipherColumn`,`plainColumn`和`assistedQueryColumn`代替逻辑列。
+
+## 14. [DistSQL] 使用DistSQL添加数据源时，如何设置自定义的JDBC连接参数或连接池属性？
+
+回答：
+
+1. 如需自定义JDBC参数，请使用 `urlSource` 的方式定义 `dataSource`。
+2. ShardingSphere预置了必要的连接池参数，如 `maxPoolSize`、`idleTimeout`等。如需增加或覆盖参数配置，请在 `dataSource` 中通过 `PROPERTIES` 指定。
+3. 以上规则请参考 [相关介绍](https://shardingsphere.apache.org/document/current/cn/features/dist-sql/syntax/rdl/rdl-resource/)
+
+## 15. [其他] 如果SQL在ShardingSphere中执行不正确，该如何调试？
+
+回答：
+
+在ShardingSphere-Proxy以及ShardingSphere-JDBC 1.5.0版本之后提供了`sql.show`的配置，可以将解析上下文和改写后的SQL以及最终路由至的数据源的细节信息全部打印至info日志。
+`sql.show`配置默认关闭，如果需要请通过配置开启。
+
+> 注意：5.x版本以后，`sql.show`参数调整为`sql-show`。
+
+## 16. [其他] 阅读源码时为什么会出现编译错误?IDEA不索引生成的代码？
+
+回答：
+
+ShardingSphere使用lombok实现极简代码。关于更多使用和安装细节，请参考[lombok官网](https://projectlombok.org/download.html)。
+
+`org.apache.shardingsphere.sql.parser.autogen` 包下的代码由 ANTLR 生成，可以执行以下命令快速生成：
+
+```bash
+./mvnw -Dcheckstyle.skip=true -Drat.skip=true -Dmaven.javadoc.skip=true -Djacoco.skip=true -DskipITs -DskipTests install -T1C 
+```
+
+生成的代码例如 `org.apache.shardingsphere.sql.parser.autogen.PostgreSQLStatementParser` 等 Java 文件由于较大，默认配置的 IDEA 可能不会索引该文件。
+可以调整 IDEA 的属性：`idea.max.intellisense.filesize=10000`
 
-## 9. 使用SQLSever和PostgreSQL时，聚合列不加别名会抛异常？
+## 17. [其他] 使用Spring命名空间时找不到xsd?
+
+回答：
+
+Spring命名空间使用规范并未强制要求将xsd文件部署至公网地址，但考虑到部分用户的需求，我们也将相关xsd文件部署至ShardingSphere官网。
+
+实际上 shardingsphere-jdbc-spring-namespace 的 jar 包中 META-INF\spring.schemas 配置了 xsd 文件的位置：
+META-INF\namespace\sharding.xsd 和 META-INF\namespace\replica-query.xsd，只需确保 jar 包中该文件存在即可。
+
+## 18. [其他] ShardingSphere除了支持自带的分布式自增主键之外，还能否支持原生的自增主键？
+
+回答：是的，可以支持。但原生自增主键有使用限制，即不能将原生自增主键同时作为分片键使用。
+
+由于ShardingSphere并不知晓数据库的表结构，而原生自增主键是不包含在原始SQL中内的，因此ShardingSphere无法将该字段解析为分片字段。如自增主键非分片键，则无需关注，可正常返回；若自增主键同时作为分片键使用，ShardingSphere无法解析其分片值，导致SQL路由至多张表，从而影响应用的正确性。
+
+而原生自增主键返回的前提条件是INSERT SQL必须最终路由至一张表，因此，面对返回多表的INSERT SQL，自增主键则会返回零。
+
```

Comment:
Should this related to `KeyGenerateAlgorithm` also use the `[sharding]` tag?

Reply:
@lanchengx Hi, the use of self-incrementing primary key cannot be used as a sharding key, I think it should not be the `[sharding]` tag.

Reply:
I think this way. 
First of all, the content of the answer is related to sharding. 
In addition, when this problem occurs, the user is probably already using the `KeyGenerateAlgorithm`.

Reply:
> I think this way.
> First of all, the content of the answer is related to sharding.
> In addition, when this problem occurs, the user is probably already using the `KeyGenerateAlgorithm`.

ok.